### PR TITLE
Prepare for the removal of the securitymanager in moxy. (#1320)

### PR DESCRIPTION
**Pull request by Tomas-Kraus**
(cherry picked from commit 88c614793706c3b55600c806c0dfeb16cdb4344f)

Signed-off-by: Tomas Kraus <tomas.kraus@oracle.com>

### CODE REVIEW

**Comment by Tomas-Kraus** [comment]
Let's wait for the tests. I did not run them locally.

File: foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/platform/server/was/WebSphere_7_Platform.java [hunk 1]
```diff
@@ -90,13 +90,13 @@ protected SessionLog createSessionLog() {
                 log = AccessController.doPrivileged(new PrivilegedExceptionAction<SessionLog>() {
                     @Override
                     public SessionLog run() throws Exception {
-                        Class<?> cls = PrivilegedAccessHelper.getClassForName(SERVER_LOG_CLASS);
+                        Class<SessionLog> cls = PrivilegedAccessHelper.getClassForName(SERVER_LOG_CLASS);
```
Comment (lukasj) [inline foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/platform/server/was/WebSphere_7_Platform.java:93]: `Class<? extends SessionLog>` would have been better

**Comment by lukasj** [comment]
```
    [junit] Running org.eclipse.persistence.testing.moxy.unit.jaxb.MOXySystemPropertiesTestCase
    [junit] Tests run: 2, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 0.334 sec
```

...and more...

File: foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/platform/server/was/WebSphere_7_Platform.java (hunk 1 above)
Comment (Tomas-Kraus) [reply foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/platform/server/was/WebSphere_7_Platform.java:93]: Yes, sure. :)

**Comment by Tomas-Kraus** [comment]
I know about failing tests. There was some jMockit test from Martin Vojtek in MOXySystemPropertiesTestCase which can't work properly after my changes. I have discussed it with Radek and modified it in a similar way he did it in 3.0/master.

**Review by lukasj — APPROVED** [review]
LGTM

**Comment by rfelcman** [comment]
Hmm it seems, that there are some test failures in
https://ci.eclipse.org/eclipselink/job/eclipselink-nightly-2.7/215/

**Comment by rfelcman** [comment]
I'll try to execute nightly build again. It should be some environment issue.

**Pull request merged**